src/Value/Color.php

```diff
@@ -282,4 +286,76 @@ private function renderAsHex(): string
 
         return '#' . ($canUseShortVariant ? $result[0] . $result[2] . $result[4] : $result);
     }
+
+    /**
+     * The "legacy" syntax does not allow RGB colors to have a mixture of `percentage`s and `number`s.
+     *
+     * The "legacy" and "modern" monikers are part of the formal W3C syntax.
+     * See the following for more information:
+     * - {@link
+     *     https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb#formal_syntax
+     *     Description of the formal syntax for `rgb()` on MDN
+     *   };
+     * - {@link
+     *     https://www.w3.org/TR/css-color-4/#rgb-functions
+     *     The same in the CSS Color Module Level 4 W3C Candidate Recommendation Draft
+     *   } (as of 13 February 2024, at time of writing).
+     */
+    private function shouldRenderInModernSyntax(): bool
+    {
+        static $colorFunctionsThatWithMixedValueTypesCannotBeRenderedInLegacySyntax = ['rgb', 'rgba'];
+        $colorFunction = $this->getRealName();
+        if (!\in_array($colorFunction, $colorFunctionsThatWithMixedValueTypesCannotBeRenderedInLegacySyntax, true)) {
+            return false;
```

Comment:
I propose we move the (static) variable and the `in_array` to a dedicated, well-named method.

Also, please generally let's not use `static` variables as they keep state across instances and are a pain to test. In this case, I think a (normal) local variable or a constant would be fine.

Reply:
> I propose we move the (static) variable and the `in_array` to a dedicated, well-named method.

Done.

> Also, please generally let's not use `static` variables as they keep state across instances and are a pain to test. In this case, I think a (normal) local variable or a constant would be fine.

Point taken.  In the new method it's not `static`.  For the time being, until these changes have 'settled down', I'd prefer not to introduce a class constant, only to remove it later.